modules/EvManager/main/car_simulatorImpl.cpp

```diff
@@ -0,0 +1,586 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#include "car_simulatorImpl.hpp"
+#include "main/SimCommand.hpp"
+#include <everest/logging.hpp>
+
+namespace module::main {
+
+void car_simulatorImpl::init() {
+    loopIntervalMs = defaultLoopIntervalMs;
+    resetSimDataToDefaults();
+    registerAllCommands();
+    subscribeToExternalMQTT();
+    subscribeToVariablesOnInit();
+}
+
+void car_simulatorImpl::ready() {
+    enabled = false;
+
+    setupEVParameters();
+
+    if (mod->config.auto_enable) {
+        auto enableCopy = mod->config.auto_enable;
+        handle_enable(enableCopy);
+    }
+    if (mod->config.auto_exec) {
+        auto valueCopy = mod->config.auto_exec_commands;
+        handle_executeChargingSession(valueCopy);
+    }
+}
+
+void car_simulatorImpl::handle_enable(bool& value) {
+    if (!enabled.has_value()) {
+        EVLOG_warning << "Already received data, but framework is not ready yet.";
+        return;
+    }
+
+    if (enabled == value) {
+        // ignore if value is the same
```

Comment:
Adding `EVLOG_warning << "Simulation is already enabled, ignoring enable!";`

Reply:
Added the log